src/fs_cell_group.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <cell_group.hpp>
+#include <fs_cell.hpp>
+#include <util/span.hpp>
+#include <util/unique_any.hpp>
+
+
+#include <iostream>
+namespace nest {
+namespace mc {
+
+/// Cell_group to collect cells that spike at a set frequency
+/// Cell are lightweight and are not executed in anybackend implementation
+class fs_cell_group : public cell_group {
+public:
+    using source_id_type = cell_member_type;
+
+    fs_cell_group(cell_gid_type first_gid, const std::vector<util::unique_any>& cells):
+        gid_base_{ first_gid }
+    {
+        using util::make_span;
+
+        auto source_gid = cell_gid_type{ gid_base_ };
+        for (auto i : make_span(0, cells.size())) {
+            // Copy all the fs_cells
+            cells_.push_back(util::any_cast<fs_cell>(cells[i]));
+
+            // create a lid to gid map
+            spike_sources_.push_back(source_id_type{ source_gid, 0 });
+            ++source_gid;
```

Comment:
Given that we already have an index `i` in this loop, why maintain a separate counter for `source_gid`? I.e., I think it would be simpler to just have
```
    spike_sources_.push_back({gid_base_+i, 0});
```
or equivalent.